core-tests/shared/src/test/scala/zio/ZIOSpec.scala

```diff
@@ -3948,6 +3948,14 @@ object ZIOSpec extends ZIOBaseSpec {
         } yield {
           assert(value)(equalTo("Controlling side-effect of function passed to promise"))
         }
+      },
+      test("onPlatform") {
+        for {
+          platform <- ZIO.suspendSucceedWith((platform, _) => ZIO.succeed(platform))
```

Comment:
Looking at these tests, probably we should introduce `ZIO.platform` to do the whole `suspendSucceedWith` thing of extracting and returning the platform, in a similar vein to `ZIO.runtime`.

Reply:
Done now. So much nicer!